packages/svelte/src/internal/client/each.js

```diff
@@ -50,7 +50,7 @@ function no_op() {}
  * @returns {void}
  */
 function each(anchor_node, collection, flags, key_fn, render_fn, fallback_fn, reconcile_fn) {
-	const is_controlled = (flags & EACH_IS_CONTROLLED) !== 0;
+	const is_controlled = !!(flags & EACH_IS_CONTROLLED);
```

Comment:
If `is_controlled` is only ever used in contexts where its truthiness matters rather than it being `true` or `false` (I don't know whether this is the case), then we could remove the `!!` as well.

Reply:
usually where I left `!!` it was to appease typescript in a quick an minimal way

Reply:
To be honest the biggest culprit for perf in my findings wasn't the boolean conversion on flags, but rather on `null` values. If you look at the bytecode generated by V8, it's almost twice as many operations needed when doing boolean conversion on these fields (interestingly). So I'd be more leaning towards having an `is_null` function there, which is readable and also very fast.

Reply:
Hmm. But the "hybrid" option from the benchmark is consistently the slowest or 2nd slowest one for me, so I'm not sure I buy that it's faster and it's certainly much harder to write

Reply:
It was the fastest for me consistently with the first one. I also find it fine to write – I don't find writing explicit code to be of any benefit, especially if the type system isn't there to help you.